src/Settings/AlgoliaSettingsManager.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Algolia\SearchBundle\Settings;
+
+
+use AlgoliaSearch\Client;
+use Symfony\Component\Filesystem\Filesystem;
+
+class AlgoliaSettingsManager implements SettingsManagerInterface
+{
+    protected $algolia;
+    protected $config;
+
+    public function __construct(Client $algolia, array $config)
+    {
+        $this->algolia = $algolia;
+        $this->config = $config;
+    }
+
+    public function backup($settingsDir, array $params)
+    {
+        $indices = $this->getIndexNames($params['indices']);
+        $fs = new Filesystem();
+        $output = [];
+
+        if (!$fs->exists($settingsDir)) {
+            $fs->mkdir($settingsDir);
+        }
+
+        foreach ($indices as $indexName) {
+            $index = $this->algolia->initIndex($indexName);
+            $settings = $index->getSettings();
+            $filename = $this->getFileName($settingsDir, $indexName, 'settings');
+
+            $fs->dumpFile($filename, json_encode($settings, JSON_PRETTY_PRINT));
+
+            $output[] = "Saved settings for <info>$indexName</info> in $filename";
+        }
+
+        return $output;
+    }
+
+    public function push($settingsDir, array $params)
+    {
+        $indices = $this->getIndexNames($params['indices']);
+        $output = [];
+
+        foreach ($indices as $indexName) {
+            $filename = $this->getFileName($settingsDir, $indexName, 'settings');
+
+            if (is_readable($filename)) {
```

Comment:
I'd rather see all the file manipulation operations (reading etc...) done via http://symfony.com/doc/current/components/finder.html, what do you think ?

Reply:
If I need to do more complex stuff I will introduce it but right now I think it's fine. Unless there are issues with OS's :(